src/XR/features/WebXRPlaneDetector.ts

```diff
@@ -93,12 +96,11 @@ export class WebXRPlaneDetector extends WebXRAbstractFeature {
     constructor(_xrSessionManager: WebXRSessionManager, private _options: IWebXRPlaneDetectorOptions = {}) {
         super(_xrSessionManager);
         this.xrNativeFeatureName = "plane-detection";
+        this._initPlaneDetector = this._initPlaneDetector.bind(this);
         if (this._xrSessionManager.session) {
-            this._init();
+            this._initPlaneDetector(this._xrSessionManager.session);
         } else {
-            this._xrSessionManager.onXRSessionInit.addOnce(() => {
-                this._init();
-            });
+            this._xrSessionManager.onXRSessionInit.addOnce(this._initPlaneDetector);

```

Comment:
Open Issue: This callback is not fired if the WebXRMeshDetector is enabled prior to the WebXRPlaneDetector when no session is currently running

Reply:
it should actually be fired. If no session is available (meaning - enabled but not in XR), the callback ias added to the XR Session Init observable. If it isn't fired, something is wrong.

Note also that there was a reason for adding the anonymous function to call this._init(), as when you add a callback directly, its context changes (so `this` is no longer what you expect it to be). There are ways around it, I find the anonymous function) the most readable. Especially when it only happens once. On each frame I would do it differently.

Reply:
@RaananW after script debugging with BabylonNative, it appears this is an issue with Observable._deferUnregister. In BabylonNative, _deferUnregister results in Observable._observers being edited during a for loop. It looks like the intention of this logic is to avoid removing observers in a synchronous manner. But for BabylonNative UWP + Chakra, this removal ends up being synchronous. 

In babylon.max.js, a for loop over _observers gets reduced to a for loop based on array length. This array length gets edited due to the synchronous removal of an observer. It causes the for loop to skip the n observers based on n previous observers being removed from the array length.

Is there an expectation that Observable._deferUnregister executes asynchronously? It seems like a lot of things may actually be broken based on this observable bug in BabylonNative.

Reply:
filed an issue here: https://github.com/BabylonJS/Babylon.js/issues/9362

Happy to help drive a fix for this problem. I'm not sure what the intended behavior is for Observable._deferUnregister

Reply:
This should now be resolved, the issue was specific to BabylonNative